utils.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"crypto/rand"
+	"io/ioutil"
+
+	p2pcrypto "github.com/libp2p/go-libp2p-crypto"
+)
+
+const ProxyIDHeader = "Proxy-ID"
+const ProxySignatureHeader = "Proxy-Signature"
+
+func getPrivateKey(path string) (p2pcrypto.PrivKey, error) {
+	if path == "" {
+		// If path is empty, generate a new key.
+		priv, _, err := p2pcrypto.GenerateSecp256k1Key(rand.Reader)
+		if err != nil {
+			return nil, err
+		}
+		return priv, nil
+	}
```

Comment:
Not sure how I feel about this. Since this is infra we will be running, and we'll need to hard-code the same private key in both locations, I'd rather we throw if a specific private key is not specified at instantiation.

Reply:
To be frank, this was simply a copy and paste, and I didn't even notice the empty path case...

Now throws an error.